### PR TITLE
Entry lock

### PR DESCRIPTION
**Pull request by kompotkot**
## Changes

Design doc: https://docs.google.com/document/d/1kA57f-Xn06VMicesQ1DB56RekbAgKIQd0DcBI2UHQL0/edit#

## How to test these changes?

Tested locally

### CODE REVIEW

**Comment by kompotkot** [comment]
@bugout-dev check
- [ ] Run migration

File: alembic/versions/5d4ec1330a98_entry_lock.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+down_revision = '245914e1ddf9'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('journal_entries', sa.Column('locked_by', sa.String(), nullable=True))
+    op.create_index(op.f('ix_journal_entries_locked_by'), 'journal_entries', ['locked_by'], unique=False)
+    op.create_unique_constraint('uq_journal_entries_id_locked_by', 'journal_entries', ['id', 'locked_by'])
```
Comment (zomglings) [inline alembic/versions/5d4ec1330a98_entry_lock.py]: `id` is primary key so it's already unique. This constraint doesn't add anything.

**Review by zomglings — APPROVED** [review]
gg

**Comment by zomglings** [comment]
Migration was taking a long time because of index creation. We cancelled the migration for now. Need to update implementation so that we store a separate table with current locks.

**Review by zomglings — APPROVED** [review]
gg

File: spire/journal/api.py [hunk 2]
```diff
@@ -1362,7 +1295,61 @@ async def update_entry_content(
...
     return JournalEntryContent(
-        title=journal_entry.title, content=journal_entry.content, tags=tags
+        title=journal_entry.title,
+        content=journal_entry.content,
+        tags=tags,
+        locked_by=entry_lock.locked_by,
+    )
+
+
+@app.delete(
```
Comment (zomglings) [inline spire/journal/api.py:1305]: Might be a good idea to make an explicit `CREATE` endpoint here, as well.

**Review by zomglings — APPROVED** [review]
gg

**Pull request merged**